app/service/auth_svc.py

```diff
@@ -52,13 +56,17 @@ async def helper(*args, **params):
 
 
 class AuthService(AuthServiceInterface, BaseService):
-
     User = namedtuple('User', ['username', 'password', 'permissions'])
 
     def __init__(self):
         self.user_map = dict()
         self.log = self.add_service('auth_svc', self)
-        self.ldap_config = self.get_config('ldap')
+        self._default_login_handler = self.DefaultLoginHandler(self.get_services())
+        self._login_handler = self._set_primary_login_handler()
```

Comment:
Why have the default and login handlers? I guess I was thinking that there would be a singular login handler (or perhaps a stack but that's likely overboard for what this needs). If someone configures a handler, it's the handler. If not, the default is the handler.

Reply:
I wanted to allow a fallback option in case the custom login handler was misconfigured for whatever reason, and users needed a backup method to log into the application. Maybe we could have the default login handler stored as a class object - that way, the auth service can have a single login handler set for itself (either the default login handler or a custom one), and the default can be referenced when needed.